File: 189-rotate-array/189. Rotate Array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        #[1,2,3,4,5,6,7,8], k=26 -> 2
        #[8,7,6,5,4,3,2,1]
        #[6,7,8,1,2,3,4,5]
        k %= len(nums)
        def reverse(start, end):
            while start <= end:
                nums[start], nums[end] = nums[end], nums[start]
                start+=1
                end-=1
        reverse(0, len(nums) - 1) #reverse entire array
        reverse(0, k-1)
        reverse(k, len(nums) - 1)


        







        


        
        

File: 189-rotate-array/189. Rotate Array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k %= nums.size();
        vector<int> lst;
        for (int i = nums.size() - k; i < nums.size(); i++) {
            lst.push_back(nums[i]);
        }
        for (int i = 0; i < nums.size() - k; i++) {
            lst.push_back(nums[i]);
        }
        for (int i = 0; i < nums.size(); i++) {
            nums[i] = lst[i];
        }
    }
};